classes/services/PKPSubmissionService.php

```diff
@@ -0,0 +1,646 @@
+ */
+
+namespace App\Services;
+
+use \DBResultRange;
+use \Application;
+use \DAOResultFactory;
+use \DAORegistry;
+
+import('lib.pkp.classes.db.DBResultRange');
+
+abstract class PKPSubmissionService {
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {}
+
+	/**
+	 * Get submissions
+	 *
+	 * @param int $contextId
+	 * @param $args array {
+	 * 		@option string orderBy
+	 * 		@option string orderDirection
+	 * 		@option int assignedTo
+	 * 		@option int|array status
+	 * 		@option string searchPhrase
+	 * 		@option int count
+	 * 		@option int offset
+	 * }
+	 *
+	 * @return array
+	 */
+	public function getSubmissionList($contextId, $args = array()) {
+
+		$defaultArgs = array(
+			'orderBy' => 'dateSubmitted',
+			'orderDirection' => 'DESC',
+			'assignedTo' => null,
+			'status' => null,
+			'searchPhrase' => null,
+			'count' => 20,
+			'offset' => 0,
+		);
+
+		$args = array_merge($defaultArgs, $args);
+
+		$submissionListQB = new QueryBuilders\SubmissionListQueryBuilder($contextId);
+		$submissionListQB
+			->orderBy($args['orderBy'], $args['orderDirection'])
+			->assignedTo($args['assignedTo'])
+			->filterByStatus($args['status'])
+			->searchPhrase($args['searchPhrase']);
+
+		\HookRegistry::call('Submission::getSubmissionList::queryBuilder', array(&$submissionListQB, $contextId, $args));
+
+		$submissionListQO = $submissionListQB->get();
+		$range = new DBResultRange($args['count'], null, $args['offset']);
+
+		$submissionDao = Application::getSubmissionDAO();
+		$result = $submissionDao->retrieveRange($submissionListQO->toSql(), $submissionListQO->getBindings(), $range);
+		$queryResults = new DAOResultFactory($result, $submissionDao, '_fromRow');
+
+		// We have to run $queryResults->toArray() before we load the next
+		// query, as it seems to interfere with the results.
+		$data = array(
+			'items' => $this->toArray($queryResults->toArray()),
+		);
+
+		$countQO = $submissionListQB->countOnly()->get();
+		$countRange = new DBResultRange($args['count'], 1);
+
+		$countResult = $submissionDao->retrieveRange($countQO->toSql(), $countQO->getBindings(), $countRange);
+		$countQueryResults = new DAOResultFactory($countResult, $submissionDao, '_fromRow');
+
+		$data['maxItems'] = (int) $countQueryResults->getCount();
+
+		return $data;
+	}
+
+	/**
+	 * Get the correct access URL for a submission's workflow based on a user's
+	 * role.
+	 *
+	 * The returned URL will point to the correct workflow page based on whether
+	 * the user should be treated as an author, reviewer or editor/assistant for
+	 * this submission.
+	 *
+	 * @param $submission Submission
+	 * @param $userId an optional user id
+	 * @param $stageName string An optional suggested stage name
+	 * @return string|false URL; false if the user does not exist or an
+	 *   appropriate access URL could not be determined
+	 */
+	public function getWorkflowUrlByUserRoles($submission, $userId = null, $stageName = null) {
+
+		$request = Application::getRequest();
```

Comment:
I'd prefer we didn't use `Application::getRequest` when possible. It's better to have the `$request` object explicitly passed down the calling hierarchy to where it's needed.